pdm/pep517/base.py

```diff
@@ -62,17 +62,36 @@ def is_same_or_descendant_path(target: str, path: str) -> bool:
 def _merge_globs(
     include_globs: Dict[str, str], excludes_globs: Dict[str, str]
 ) -> Tuple[List[str], List[str]]:
-    includes, excludes = [], []
+    """Correctly merge includes and excludes.
+    When a pattern exists in both includes and excludes,
+    determine the priority in the following ways:
+
+    1. The one with more parts in the path wins
+    2. If part number is equal, concrete file path wins
+    3. If both are of the same pattern type, **excludes** wins
```

Comment:
The implementation of this function does not fit with the docstring. It does not compare first the number of parts. And the implementations of step 2 and 3 seem weird and not clear to me.

It looks like checking if the "include" glob pattern is a concrete path first, and then checking the number of parts. The step 3 appears not implemented at all.

Reply:
`continue` means `excludes` wins and `del exclude_globs[path]` means `includes` wins.

In the first if branch:
```python
if "*" in key
```
The include pattern is a wildcard pattern, then it loses if it is shorter or equal than excludes, no matter if exclude pattern contains wildcards because even if they are of equal length, excludes or concrete path wins:
```python
if include_path_parts <= exclude_path_parts:
```
Else, include pattern is a concrete file path, and it loses if it is shorter or equally long as excludes and exclude pattern is not concrete.

Reply:
It is hard to get the concrete level of un-recursive glob patterns because there are many kinds of wildcards for the glob patterns. Only `"*" in key` is not enough to determine if the key is with wildcard.